### DISCUSSION TITLE
pydeck on_view_state_change debounce capability

### DISCUSSION THREAD

**Discussion by cornhundred** [visgl/deck.gl#5838] [Q&A] [answered]
Hi, is there a way to 'debounce' the on_view_state_change function in pydeck to prevent the user from making too many consecutive calls? This can cause the visualization to be less responsive if the python function that is called takes a while to run.

**Comment by ajduberstein** [comment, marked as answer]
There is a `debounce_seconds` keyword argument–

```
    def on_view_state_change(self, callback, debouce_seconds=0.2, remove=False):
        callback = debounce(debouce_seconds)(callback) if debouce_seconds > 0 else callback
        self._view_state_handlers.register_callback(callback, remove=remove)
```

Hopefully works-let me know.

**Reply by cornhundred** [reply]
Sorry I didn't see this :) I'll let you know how it works

**Reply by cornhundred** [reply]
it works, thanks again
